src/Functions/byteSize.cpp

```diff
@@ -0,0 +1,281 @@
+            case TypeIndex::UInt128:    byte_size = ByteSizeForNative<UInt128>::value; break;
+            case TypeIndex::UInt256:    byte_size = ByteSizeForNative<UInt256>::value; break;
+            case TypeIndex::Int8:       byte_size = ByteSizeForNative<Int8>::value; break;
+            case TypeIndex::Int16:      byte_size = ByteSizeForNative<Int16>::value; break;
+            case TypeIndex::Int32:      byte_size = ByteSizeForNative<Int32>::value; break;
+            case TypeIndex::Int64:      byte_size = ByteSizeForNative<Int64>::value; break;
+            case TypeIndex::Int128:     byte_size = ByteSizeForNative<Int128>::value; break;
+            case TypeIndex::Int256:     byte_size = ByteSizeForNative<Int256>::value; break;
+            case TypeIndex::Float32:    byte_size = ByteSizeForNative<Float32>::value; break;
+            case TypeIndex::Float64:    byte_size = ByteSizeForNative<Float64>::value; break;
+            case TypeIndex::Date:       byte_size = ByteSizeForNative<DataTypeDate>::value; break;
+            case TypeIndex::DateTime:   byte_size = ByteSizeForNative<DataTypeDateTime>::value; break;
+            case TypeIndex::DateTime64: byte_size = ByteSizeForNative<DataTypeDateTime64>::value; break;
+            case TypeIndex::Enum8:      byte_size = ByteSizeForNative<DataTypeEnum8>::value; break;
+            case TypeIndex::Enum16:     byte_size = ByteSizeForNative<DataTypeEnum16>::value; break;
+            case TypeIndex::Decimal32:  byte_size = ByteSizeForNative<Decimal32>::value; break;
+            case TypeIndex::Decimal64:  byte_size = ByteSizeForNative<Decimal64>::value; break;
+            case TypeIndex::Decimal128: byte_size = ByteSizeForNative<Decimal128>::value; break;
+            case TypeIndex::Decimal256: byte_size = ByteSizeForNative<Decimal256>::value; break;
+            case TypeIndex::UUID:       byte_size = ByteSizeForNative<DataTypeUUID>::value; break;
+            // case TypeIndex::Interval:            internal use only.
+            // case TypeIndex::Set:                 internal use only.
+            // case TypeIndex::Function:            internal use only.
+            // case TypeIndex::AggregateFunction:   internal use only.
+            default: return false;
+        }
+
+        return true;
+    }
+
+    static bool byteSizeByColumn(const IDataType * data_type, const IColumn * column, ColumnUInt64::Container & vec_res)
+    {
+        WhichDataType which(data_type);
+        size_t vec_size = vec_res.size();
+
+        if (which.isString()) // TypeIndex::String
+        {
+            const ColumnString * col_str = checkAndGetColumn<ColumnString>(column);
+            const auto & offsets = col_str->getOffsets();
+            ColumnString::Offset prev_offset = 0;
+            for (size_t i = 0; i < vec_size; ++i)
+            {
+                vec_res[i] += offsets[i] - prev_offset + sizeof(offsets[0]);
```

Comment:
One minor issue with type aliasing.
As vec_res and offsets have the same type, compiler cannot rely to the fact that they are not aliased.
It will do extra load of `offsets[i]` from memory.

To solve, you can save `offsets[i]` to the temporary variable on stack.